### PR TITLE
Fix population of $_POST from content-type=multipart/form-data in CGI mode

### PR DESCRIPTION
**Pull request by livecodefraser**

### CODE REVIEW

File: engine/src/srvcgi.cpp [hunk 1]
```diff
@@ -1383,6 +1390,20 @@ static bool cgi_store_form_multipart(IO_handle p_stream)
...
+	MCAutoArrayRef t_post_array, t_post_binary_array;
+	if (t_success)
+	{
+		t_success = MCArrayCreateMutable(&t_post_array) &&
+			MCArrayCreateMutable(&t_post_binary_array);
+			
+		if (t_success)
+		{
+			t_context.post_variable = *t_post_array;
+			t_context.post_binary_variable = *t_post_binary_array;
```
Comment (peter-b) [inline engine/src/srvcgi.cpp:1403]: Why not use `MCValueAssign()` here?

**Comment by peter-b** [comment]
I'm sure this is fine despite the slight ugliness of having `cgi_multipart_context_t` contain `MCValueRef` pointers without incrementing the reference count.

@livecode-vulcan review ok 97f3885

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 97f388565b01e865f705c0d42296dec064d2c4b3

**Comment by livecode-vulcan** [comment]
😎 test success 97f388565b01e865f705c0d42296dec064d2c4b3
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2246)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2254)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2239)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2255)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2261)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2267)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2277)

**Pull request merged**